src/renderers/dom/shared/__tests__/ReactRenderDocument-test.js

```diff
@@ -235,14 +239,19 @@ describe('rendering React components at document', () => {
       }
     }
 
-    expect(function() {
-      ReactDOM.render(<Component />, container);
-    }).toThrowError(
-      "You're trying to render a component to the document but you didn't " +
-        "use server rendering. We can't do this without using server " +
-        'rendering due to cross-browser quirks. See ' +
-        'ReactDOMServer.renderToString() for server rendering.',
-    );
+    if (ReactDOMFeatureFlags.useFiber) {
+      ReactDOM.render(<Component text="Hello world" />, testDocument);
```

Comment:
I'm not sure those cross-browser quirks are still in modern browsers. I didn't find any issues (that aren't also issues for SSR).

I'm not sure how restrictive we want to be?